std/algorithm/iteration.d

```diff
@@ -836,7 +838,7 @@ template each(alias pred = "a")
 
     enum isForeachBinaryIterable(R) =
         is(typeof((R r) {
-            foreach (i, ref a; r)
+            foreach (ref i, ref a; r)
```

Comment:
The intent was to support a situation where there actually _are_ two ref args (e.g. using `lockstep` -- see the [second commit](https://github.com/rcorre/phobos/commit/9fd7ae3927b39bb087d2b5bfbca6cb2b2dcbaff9).

But yeah, this might be an issue. I'll check if its possible to distinguish between that and an implicit loop counter.